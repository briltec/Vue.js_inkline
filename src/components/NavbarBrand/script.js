import AttributesProviderMixin from '@inkline/inkline/mixins/components/providers/AttributesProviderMixin';

export default {
    name: 'INavbarBrand',
    mixins: [
        AttributesProviderMixin
    ],
    computed: {
        isTag () {
            return this.attributes.to ? 'router-link' : 'a';
        }
    }
};