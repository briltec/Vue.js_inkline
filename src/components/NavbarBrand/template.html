<a class="brand"
     tag="a"
     :is="isTag"
     :class="classes['*']"
     v-bind="attributes">
    <slot></slot>
</a>