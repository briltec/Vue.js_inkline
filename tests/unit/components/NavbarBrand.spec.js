import { shallowMount } from '@vue/test-utils';
import NavbarBrand from 'inkline/components/NavbarBrand';
import { RouterLink } from '../mocks/RouterLink';

describe('Components', () => {
    describe('NavbarBrand', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallowMount(NavbarBrand);
        });

        it('should be named correctly', () => {
            expect(NavbarBrand.name).toEqual('INavbarBrand');
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('isTag()', () => {
                it('should be defined', () => {
                    expect(wrapper.vm.isTag).toBeDefined();
                });

                it('should be <a> tag by default', () => {
                    expect(wrapper.vm.isTag).toEqual('a');
                });

                it('should be <router-link> tag if "to" attribute is provided', () => {
                    wrapper = shallowMount(NavbarBrand, {
                        components: {
                            RouterLink
                        },
                        propsData: {
                            to: true
                        }
                    });

                    expect(wrapper.vm.isTag).toEqual('router-link');
                });
            });
        });
    });
});
