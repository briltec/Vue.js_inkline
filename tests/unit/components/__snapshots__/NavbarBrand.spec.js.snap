// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components NavbarBrand should render correctly 1`] = `
<div class="alert">
  <!---->
  <div class="content"></div>
  <!---->
</div>
`;
